### PR TITLE
MGMT-4850 set InstallationDiskID based on RootDeviceHints

### PR DESCRIPTION
**Pull request by flaper87**
This PR adds the ability to select an installation disk based on the RootDeviceHints provided by the BMH. Note that only the hints supported by OpenShift (BareMetal Operator, specifically) were implemented here. The rest of the filters and operations supported by ironic are not needed for now as they are not even used today in the OpenShift case.

Signed-off-by: Flavio Percoco <flavio@redhat.com>

### CODE REVIEW

File: internal/controller/controllers/bmh_agent_controller.go [hunk 1]
```diff
@@ -384,6 +391,91 @@ func (r *BMACReconciler) reconcileBMH(ctx context.Context, bmh *bmh_v1alpha1.Bar
...
+		if hints.Vendor != "" && !strings.Contains(disk.Vendor, hints.Model) {
+			continue
+		}
+
+		if hints.SerialNumber != "" && hints.SerialNumber != disk.Serial {
+			continue
+		}
+
+		if hints.MinSizeGigabytes != 0 {
+			sizeGB := int(disk.SizeBytes / (1024 * 3))
```
Comment (avishayt) [inline internal/controller/controllers/bmh_agent_controller.go:441]: Use pkg/conversions/conversions.go, need to add BytesToGB

Comment (flaper87) [reply internal/controller/controllers/bmh_agent_controller.go:442]: > Use pkg/conversions/conversions.go, need to add BytesToGB

There's one `BytesToGiB`

Comment (avishayt) [reply internal/controller/controllers/bmh_agent_controller.go:442]: Disks use base 10 (GB), not base 2 (GiB).  Sorry I didn't make that clear.

**Review by avishayt — APPROVED** [review]

File: internal/controller/controllers/bmh_agent_controller.go [hunk 2]
```diff
@@ -270,22 +276,24 @@ func (r *BMACReconciler) reconcileAgentInventory(bmh *bmh_v1alpha1.BareMetalHost
...
+			Rotational:   strings.EqualFold(d.DriveType, "hdd"),
 		}
 
 		hardwareDetails.Storage = append(hardwareDetails.Storage, disk)
 	}
 
 	// Add the memory information in MebiByte
 	if agent.Status.Inventory.Memory.PhysicalBytes > 0 {
-		hardwareDetails.RAMMebibytes = int(inventory.Memory.PhysicalBytes / (1024 * 1024))
+		hardwareDetails.RAMMebibytes = int(conversions.BytesToGiB(inventory.Memory.PhysicalBytes))
```
Comment (dtantsur) [inline internal/controller/controllers/bmh_agent_controller.go:296]: MiB on the left side, GiB on the right, is it intended? I think the earlier version used MiB as well.

File: internal/controller/controllers/bmh_agent_controller.go [hunk 3]
```diff
@@ -384,6 +392,91 @@ func (r *BMACReconciler) reconcileBMH(ctx context.Context, bmh *bmh_v1alpha1.Bar
...
+			continue
+		}
+
+		if hints.SerialNumber != "" && hints.SerialNumber != disk.Serial {
+			continue
+		}
+
+		if hints.MinSizeGigabytes != 0 {
+			sizeGB := int(disk.SizeBytes / (1024 * 3))
+			if hints.MinSizeGigabytes < sizeGB {
```
Comment (dtantsur) [inline internal/controller/controllers/bmh_agent_controller.go:443]: This logic seems reversed to me.. Or am I just too sleepy?

Comment (zaneb) [reply internal/controller/controllers/bmh_agent_controller.go:443]: Nope, this is definitely backwards.

Comment (zaneb) [reply internal/controller/controllers/bmh_agent_controller.go:443]: https://bugzilla.redhat.com/show_bug.cgi?id=2131017
https://github.com/openshift/assisted-service/pull/4452

File: internal/controller/controllers/bmh_agent_controller.go (hunk 2 above)
Comment (zaneb) [reply internal/controller/controllers/bmh_agent_controller.go:296]: https://github.com/openshift/assisted-service/pull/4453

**Pull request merged**